python-sdk/src/astro/sql/operators/base_decorator.py

```diff
@@ -86,6 +86,14 @@ def execute(self, context: Context) -> None:
                 raise ValueError("You need to provide a table or a connection id")
         self.database_impl = create_database(self.conn_id, first_table)
```

Comment:
Can we move this line below the if condition, so that we avoid another DB call with that method in case tables do not belong to the same datasource?